pkg/service/credential/service.go

```diff
@@ -146,6 +149,74 @@ func (s Service) CreateCredential(request CreateCredentialRequest) (*CreateCrede
 		return nil, util.LoggingErrorMsg(err, errMsg)
 	}
 
+	if request.Revocable == true {
+		credID := builder.ID
+		issuerID := request.Issuer
+		schemaID := request.JSONSchema
+
+		storedStatusListCreds, err := s.storage.GetStatusListCredentialsByIssuerAndSchema(issuerID, schemaID)
+		if err != nil {
+			return nil, util.LoggingNewErrorf("problem with getting status list credential for issuer: %s schema: %s", request.Issuer, request.JSONSchema)
+		}
+
+		// This should never happen, there should always be only 1 status list credential per <issuer,schema> pair
+		if len(storedStatusListCreds) > 1 {
+			return nil, util.LoggingNewErrorf("only one status list credential per <issuer,schema> pair allowed. issuer: %s schema: %s", request.Issuer, request.JSONSchema)
+		}
+
+		var statusListCredentialID string
+
+		// First time that this <issuer,schema> pair has a revocation or suspension credential issued
+		if len(storedStatusListCreds) == 0 {
+			generatedStatusListCredential, err := statussdk.GenerateStatusList2021Credential(fmt.Sprintf("%s/v1/credentials/status/%s", s.config.ServiceEndpoint, uuid.New().String()), issuerID, statussdk.StatusRevocation, []credential.VerifiableCredential{})
```

Comment:
may want to break this into a new helper method `createNewStatusListCredential` or similar